DNN Platform/Tests/DotNetNuke.Tests.Core/Security/PortalSecurity/PortalSecurityTest.cs

```diff
@@ -0,0 +1,52 @@
+﻿using NUnit.Framework;
+using DotNetNuke.Security;
+
+
+namespace DotNetNuke.Tests.Core.Security.PortalSecurity
+{
+    [TestFixture]
+    public class PortalSecurityTest
+    {
+        #region Setup
+        [SetUp]
+        public void Setup()
+        {
+            //TODO: implement environemnt set up
+        }
+
+        #endregion
+
+        #region Tear Down
+        [TearDown]
+        public void TearDown()
+        {
+            //TODO:implement clean up
+        }
+        #endregion
+
+        #region Test Methods
+        [Test]
```

Comment:
We can use **TestCase** to easily test different scenarios. Like this:
```
[TestCase("<source></source>", "")]
[TestCase("<source>Hi this is personal Test for source tag</source>   ", "   ")]
[TestCase("<SOURCE>Upper case</SOURCE>", "")]
[TestCase("<source src="https://google.com/">Source with attribute</source>", "")]
[Add more test cases here]
public void Typing_Html_Source_Tag_Should_Not_Be_Allowed(string html, string expectedOutput)
{
        // Use html parameter as input and expectedOutput to assert the result
}
```

Reply:
OMHO, **Typing_** prefix is not necessary in the name. **Html_Source_Tag_Should_Not_Be_Allowed** is better.